picoscope/ps5000a.py

```diff
@@ -274,14 +287,15 @@ def _lowLevelSetSimpleTrigger(self, enabled, trigsrc, threshold_adc,
         self.checkResult(m)
 
     def _lowLevelRunBlock(self, numPreTrigSamples, numPostTrigSamples,
-                          timebase, oversample, segmentIndex):
+                          timebase, oversample, segmentIndex, callback,
+                          pParameter):
         # Oversample is NOT used!
         timeIndisposedMs = c_int32()
         m = self.lib.ps5000aRunBlock(
             c_int16(self.handle), c_uint32(numPreTrigSamples),
             c_uint32(numPostTrigSamples), c_uint32(timebase),
             byref(timeIndisposedMs), c_uint32(segmentIndex),
-            c_void_p(), c_void_p())
+            callback, pParameter)
```

Comment:
```suggestion
            blockReady(callback), pParameter)
```

Is this not more in the spirit of these low level py files?

Reply:
You're right, that would be better. It should work that way as he decorotator doesn't do it any different. I'll try it.

Reply:
Using blockReady(callback) instead of the decorator does not work.

The reason might be, that according to https://docs.python.org/3/library/ctypes.html#callback-functions the ctype function has to remain in memory the whole time. Which is the case, if the function is decorated. But in the case of 'blockReady(callback)' it is not stored as a ctype function.